File: Coding Challenge/C++/Question2.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(string files) 
{
    map<char, int> count;
    for (char c : files) {
        counter[c]++;
    }
    int s = 0;
    int t = 0;
    for (auto p : count) {
        t += (p.second & ~1);
        if (p.second & 1 && !s) {
            s = 1;
        }
    }

    return t+s;
}


int main() {
     string files="";
     cin>>files;
     cout<<solution(files);
     return 0;
}

File: Coding Challenge/Python/Question2.py
from collections import Counter


def solution(files):
    myCounter = Counter()
    total = 0
    spare = 0
    for char in files:
        myCounter[char] += 1
        if (myCounter[char] & 1):
            spare += 1
        else:
            total += 2
            spare -= 1
    if spare:
        return total + 1
    return total


def main():
    str = input()
    answer = solution(str)
    print(answer)


if __name__ == '__main__':
    main()
